algebird-test/src/test/scala/com/twitter/algebird/BytesSpec.scala

```diff
@@ -0,0 +1,89 @@
+package com.twitter.algebird
+
+import org.scalacheck.Gen
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+import org.scalatest.{Matchers, WordSpec}
+
+class BytesSpec extends WordSpec with Matchers with GeneratorDrivenPropertyChecks {
+
+  "has a sane equals function" in {
+    val random = new scala.util.Random
+    forAll((Gen.choose(1, 4096), "wordLength")) { (wordLength: Int) =>
+      // Given two different array instances that have the same content.
+      val word = random.nextString(wordLength)
```

Comment:
I think we need some tests to exercise the case that they are not all equal. Right now, equals(t: Any) == (t != null) and compare(a, b) == 0 would pass your tests